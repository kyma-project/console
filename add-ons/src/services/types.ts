import { Filters, ConfigurationLabels, Configuration } from '../types';

export interface FiltersService {
  activeFilters: Filters;
  setFilterLabel: (label: string) => void;
  removeFilterLabel: (label: string) => void;
  removeAllFilters: () => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface LabelsService {
  getUniqLabels: ConfigurationLabels;
}

export interface ConfigurationsService {
  originalConfigs: Configuration[];
  filteredConfigs: Configuration[];
}

export enum SubscriptionType {
  ADD = 'ADD',
  UPDATE = 'UPDATE',
  DELETE = 'DELETE',
}
