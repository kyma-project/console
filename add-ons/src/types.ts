import React from 'react';

export type BackendModule = string;

export interface SecretRef {
  name: string;
  namespace: string;
}

export interface Repository {
  url: string;
  secretRef?: SecretRef;
}

export interface RepositoryAddon {
  name: string;
  status: string;
  version: string;
  reason: string;
  message: string;
}

export interface RepositoryStatus {
  url: string;
  status: string;
  reason: string;
  message: string;
  addons: RepositoryAddon[];
}

export interface ConfigurationStatus {
  phase: string;
  repositories: RepositoryStatus[];
}

export interface Configuration {
  name: string;
  urls: ConfigurationURL[];
  labels: ConfigurationLabels;
  repositories: Repository[];
  status: ConfigurationStatus;
}

export interface ConfigurationLabels {
  [key: string]: string;
}
export type ConfigurationURL = string;

export interface FilterLabels {
  [key: string]: string[];
}

export interface Filters {
  search: string;
  labels: FilterLabels;
}

export interface ActiveFiltersAction {
  payload: {
    key: string;
    value: string;
  };
  type: ActiveFiltersActionType;
}

export enum ActiveFiltersActionType {
  SET_SEARCH = 'SET_SEARCH',
  SET_LABEL = 'SET_LABEL',
  REMOVE_LABEL = 'REMOVE_LABEL',
  REMOVE_ALL_FILTERS = 'REMOVE_ALL_FILTERS',
}

export interface RemoveFiltersInterface {
  removeFilterLabel: (key: string, value: string) => void;
  removeAllFiltersLabels: () => void;
}

export interface FiltersLabelsInterface {
  uniqueLabels: FilterLabels;
  setFilterLabel: (key: string, value: string) => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface ConfigurationsService
  extends FiltersLabelsInterface,
    RemoveFiltersInterface {}

export interface Notification {
  title: string;
  content: string | React.ReactNode;
  color?: string;
  icon?: string;
}
