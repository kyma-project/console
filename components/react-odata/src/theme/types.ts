export interface ThemeInterface {
  color?: string;
}
