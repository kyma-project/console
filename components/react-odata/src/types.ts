export interface Child {
  attributes: { [key: string]: string };
  name: string;
  value: string;
  children: Child[];
}
