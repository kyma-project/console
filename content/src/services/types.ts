import { ClusterAssetGroup } from '@kyma-project/common';

export interface ClusterAssetGroups {
  [group: string]: ClusterAssetGroup[];
}

export interface Navigation {
  [group: string]: NavigationItem[];
}

export interface NavigationItem {
  name: string;
  group: string;
  displayName: string;
}
