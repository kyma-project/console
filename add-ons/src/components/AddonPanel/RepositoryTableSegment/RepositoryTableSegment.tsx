import React from 'react';
import { Button, Icon, Popover } from 'fundamental-react';
import DeleteUrlModal from './../../Modals/DeleteUrlModal/DeleteUrlModal.container';
import { RepositoryStatus, RepositoryAddon } from '../../../types';
import classNames from 'classnames';

interface RepositoryTableHeaderProps {
  isOpen: boolean;
  setIsOpen: (state: boolean) => void;
  repository: RepositoryStatus;
  configName: string;
}

const RepositoryTableHeader: React.FunctionComponent<
  RepositoryTableHeaderProps
> = ({ isOpen, setIsOpen, repository, configName }) => (
  <tr>
    <th>
      <Button
        glyph={isOpen ? 'navigation-up-arrow' : 'navigation-down-arrow'}
        option="light"
        onClick={() => setIsOpen(!isOpen)}
        disabled={!repository.addons.length}
      />
    </th>
    <th>{repository.url}</th>
    <th />
    <th>
      <StatusWithMessage
        status={repository.status}
        message={repository.message}
      />
    </th>
    <th>
      {<DeleteUrlModal url={repository.url} configurationName={configName} />}
    </th>
  </tr>
);

interface RepositoryTableRowProps {
  addon: RepositoryAddon;
}

const RepositoryTableRow: React.FunctionComponent<RepositoryTableRowProps> = ({
  addon,
}) => (
  <tr>
    <td />
    <td className="addon-table-padded">{addon.name}</td>
    <td>{addon.version}</td>
    <td>
      <StatusWithMessage status={addon.status} message={addon.message} />
    </td>
    <td />
  </tr>
);

interface StatusWithMessageProps {
  status: string;
  message: string;
}

const StatusWithMessage: React.FunctionComponent<StatusWithMessageProps> = ({
  status,
  message,
}) => {
  const statusClassName = classNames({
    'fd-has-color-status-1': status === 'Ready',
    'fd-has-color-status-3': status === 'Failed',
  });

  const body = (
    <div className="fd-has-padding-xs has-max-width-m has-white-space-normal has-word-break-break-word">
      {message}
    </div>
  );

  const icon = (
    <Icon
      glyph="error"
      size="s"
      className="fd-has-color-status-3 fd-has-margin-left-tiny"
    />
  );

  return (
    <>
      {<span className={statusClassName}>{status}</span>}
      {message && <Popover body={body} control={icon} placement="bottom-end" />}
    </>
  );
};

interface RepositoryTableSegmentProps {
  repository: RepositoryStatus;
  configName: string;
}

const RepositoryTableSegment: React.FunctionComponent<
  RepositoryTableSegmentProps
> = ({ repository, configName }) => {
  const [isOpen, setIsOpen] = React.useState(false);

  return (
    <tbody>
      <RepositoryTableHeader
        isOpen={isOpen}
        setIsOpen={setIsOpen}
        repository={repository}
        configName={configName}
      />
      {isOpen &&
        repository.addons.map((addon: RepositoryAddon) => (
          <RepositoryTableRow key={addon.name} addon={addon} />
        ))}
    </tbody>
  );
};

export default RepositoryTableSegment;
