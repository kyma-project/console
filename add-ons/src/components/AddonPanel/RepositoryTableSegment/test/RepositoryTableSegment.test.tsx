import React from 'react';
import RepositoryTableSegment from './../RepositoryTableSegment';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RepositoryStatus } from '../../../../types';

const repositoryMock: RepositoryStatus = {
  url: 'url',
  status: 'ready',
  reason: '',
  message: '',
  addons: [],
};

describe('RepositoryTableSegment', () => {
  it('Renders with minimal props', () => {
    const component = shallow(
      <RepositoryTableSegment
        repository={repositoryMock}
        configName="config name"
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
