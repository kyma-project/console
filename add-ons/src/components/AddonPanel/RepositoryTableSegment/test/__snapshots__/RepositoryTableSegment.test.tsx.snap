// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTableSegment Renders with minimal props 1`] = `
<tbody>
  <RepositoryTableHeader
    configName="config name"
    isOpen={false}
    repository={
      Object {
        "addons": Array [],
        "message": "",
        "reason": "",
        "status": "ready",
        "url": "url",
      }
    }
    setIsOpen={[Function]}
  />
</tbody>
`;
